Skips stacking if no valid mask_safe (#6621)
gammapy/gammapy: docs/release-notes/6621.bug.rst

> Skips stacking if no valid mask_safe, fix gti preventing stacking of invalid dataset

Reviewer: ```suggestion
`~gammapy.datasets.MapDataset.stack(other)` now skips stacking if other mask_safe is invalid. This prevents stacking gti of invalid dataset.
```
